source/settings/plugin/mail-crypt-plugin.rst

```diff
@@ -1,31 +1,130 @@
+.. seealso:: See :ref:`mail_crypt_plugin` for an overview.
 
-In case of unauthorized access to the storage backend, the messages will, without access to the decryption keys, be unreadable to the offending party.
+Settings
+========
+
+.. _setting-plugin_mail_crypt_acl_require_secure_key_sharing:
+
+``mail_crypt_acl_require_secure_key_sharing``
+---------------------------------------------
+
+- Default: ``0``
+- Values:  :ref:`string`
+
+If true (``1`` or ``y``), require secure key sharing.
 
-There can be a single encryption key for the whole system or each user can have a key of their own. The used cryptographical methods are widely used standards and keys are stored in portable formats, when possible.
 
 .. _setting-plugin_mail_crypt_curve:
 
 ``mail_crypt_curve``
----------------------
+--------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Defines the elliptic curve to use for key generation.
+
+Any valid curve supported by the underlying cryptographic library is allowed.
+
+Example::
+
+  plugin {
+    mail_crypt_curve = secp521r1
+  }
+
+This must be set if you wish to use folder keys rather than global keys.
+
+With global keys (either RSA or EC keys), all keying material is taken from the
+plugin environment and no key generation is performed.
+
+In folder-keys mode, a key pair is generated for the user, and a
+folder-specific key pair is generated. The latter is encrypted by means of the
+user's key pair.
+
+
+.. _setting-plugin_mail_crypt_global_public_key:
+
+``mail_crypt_global_public_key``
+--------------------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Public key to encrypt files. Key must be in PEM format, using pkey format.
+
+
+.. _setting-plugin_mail_crypt_global_private_key:
+
+``mail_crypt_global_private_key``
+---------------------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Private key(s) to decrypt files. Key(s) must be in PEM format, using pkey
+format.
+
+You can define multiple keys by appending an increasing number to the setting
+label.
+
+
+.. _setting-plugin_mail_crypt_private_key:
+
+``mail_crypt_private_key``
+--------------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Private key to decrypt user's master key. Key must be in PEM format, using
+pkey format.
+
+
+.. _setting-plugin_mail_crypt_private_password:
+
+``mail_crypt_private_password``
+-------------------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Password to decrypt user's master key or environment private key.
+
+
+.. _setting-plugin_mail_crypt_require_encrypted_user_key:
+
+``mail_crypt_require_encrypted_user_key``
+-----------------------------------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+If true (any value, such as ``1``, will enable), require user key encryption
+with password.
```

Comment:
blah.. this is probably the only such plugin {} setting left that isn't a proper boolean. Lets fix this?

Also, somewhere in this pull request I saw you describing similar thing as "Values: (existence - any value means yes)" or something like that. That sounds nicer.